### PR TITLE
Use `x11rb` features to improve code

### PR DESCRIPTION
**Pull request by daxpedda**
I was trying to learn from your example to improve on ArturKovacs/arboard#9 when I saw you playing around with `to_le_bytes`.

### CODE REVIEW

File: x11/src/lib.rs [hunk 1]
```diff
@@ -396,8 +396,8 @@ impl Worker {
                             &data,
                         )
                         .expect("Change property");
-                    } else if value.len() < max_length - 24 {
-                        let _ = self.context.connection.change_property8(
+                    } else if value.len() < max_length {
+                        let _ = self.context.connection.change_property32(
```
Comment (daxpedda) [inline x11/src/lib.rs]: I believe changing this to `u32` simplified this, I am unsure though if it is also correct.

**Comment by hecrj** [comment]
If I run the `write` example in this branch, I get an incorrect result when pasting:

```
H���e���l���l���o���,��� ���w���o���r���l���d���!���
```

File: x11/src/lib.rs (hunk 1 above)
Comment (daxpedda) [reply x11/src/lib.rs]: So apparently this doesn't work.
My apologies for not noticing this very obvious flaw.

**Comment by daxpedda** [comment]
Reverting the `u32` change, the rest should work.

**Comment by daxpedda** [comment]
I am trying to figure out why you did some of the calculations here:

The X11 spec defines that the length returned by the Big Request extension is display in 4-byte units:
https://www.x.org/releases/X11R7.6/doc/bigreqsproto/bigreq.html#c_language_binding
> [..] otherwise returns the maximum-request-length (in 4-byte units) supported by the server through the extended-length encoding.

https://github.com/hecrj/window_clipboard/blob/eec369da574e00eee5a9a072fa75bb9143fa0727/x11/src/lib.rs#L365
I am assuming that this is why the length is multiplied by `4`, to get the actualy size in single bytes.

But as far as I understand [`maximum_request_bytes`](https://docs.rs/x11rb/0.8.0/x11rb/connection/trait.RequestConnection.html#tymethod.maximum_request_bytes) is already returning the numbers of single bytes correctly:
https://github.com/psychon/x11rb/blob/424ca2bc4aa0cfff628d5d420aa8cec6afa50fc1/src/rust_connection/mod.rs#L595

Further below is this:
https://github.com/hecrj/window_clipboard/blob/eec369da574e00eee5a9a072fa75bb9143fa0727/x11/src/lib.rs#L412
I'm completely unsure what the `- 24` does.

**Comment by hecrj** [comment]
Not completely sure about those. I mostly ported #3 and #9 to `x11rb` without changing any logic.

- The multiplication by `4` does seem unnecessary.
- Could the `24` constant be the constant size of the change property request?

**Comment by daxpedda** [comment]
I did some digging, as far as I can see we don't need to account for any of this because `x11rb` does this already for us:
`change_property` eventually leads to:
https://github.com/psychon/x11rb/blob/424ca2bc4aa0cfff628d5d420aa8cec6afa50fc1/src/rust_connection/mod.rs#L207-L234

which uses `compute_length_field`:
https://github.com/psychon/x11rb/blob/424ca2bc4aa0cfff628d5d420aa8cec6afa50fc1/src/connection.rs#L488-L559

`compute_length_field` basically checks if the size of the to-be-sended message is to big, requests Big Request if it is, and even splits up the message into multiple ones if it still doesn't fit.

Again, I am a beginner when it comes to X11, I will do some experiments.

**Comment by daxpedda** [comment]
I played around with 100 MB files, everything seems to be working, but I will do that in a separate PR.
Are there any other concerns with this one?

**Review by hecrj — APPROVED** [review]
Looks good! Thank you :beers:

**Comment by psychon** [comment]
> Could the 24 constant be the constant size of the change property request?

Yup. Let's look at `/usr/include/xcb/xproto.h`. We find:
```c
typedef struct xcb_change_property_request_t {
    uint8_t      major_opcode;
    uint8_t      mode;
    uint16_t     length;
    xcb_window_t window;
    xcb_atom_t   property;
    xcb_atom_t   type;
    uint8_t      format;
    uint8_t      pad0[3];
    uint32_t     data_len;
} xcb_change_property_request_t;
```
Knowing that `xcb_window_t` and `xcb_atom_t` are four bytes large, I conclude that this request has size 24. "Trust me, I know what I am doing" says that the actual data comes directly after the above fixed-size header.

**Pull request merged**